src/Control/Eff/Cut.hs

```diff
@@ -2,6 +2,8 @@
 {-# LANGUAGE PatternGuards #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
```

Comment:
@qnikst could you elaborate on why this is needed? i.e. what is the behaviour in ghc-7.10 without ``ScopedTypeVariables`` (and similarly for other changes in this commit). feel free to point to relevant portions of ghc-7.10 changelog which talk about issues which necessitate this